x-pack/test/smb-fixture/src/main/resources/certs/README.asciidoc

```diff
@@ -0,0 +1,52 @@
+
+=== Generate certificates for the Samba server
+
+These keys and certificates are copied by installsmb.sh for the Samba server to use for TLS. See `../resources/installsmb.sh`
+
+[source,shell]
+-----------------------------------------------------------------------------------------------------------
+elasticsearch-certutil ca --pem --out=${PWD}/samba-ca.zip
+
+unzip samba-ca.zip
+mv ca/ca.crt ./ca.pem
+mv ca/ca.key ./ca.key
+
+rm samba-ca.zip
+rm -r ca
+-----------------------------------------------------------------------------------------------------------
+
+[source,shell]
+-----------------------------------------------------------------------------------------------------------
+elasticsearch-certutil cert  --pem --ca-cert ${PWD}/ca.pem --ca-key ${PWD}/ca.key \
+   --dns localhost --ip 127.0.0.1,0:0:0:0:0:0:0:1 --name samba4
+
+unzip samba4.zip
+mv samba4/samba4.crt ./cert.pem
+mv samba4/samba4.key ./key.pem
+
+rm samba4.zip
+rm -r samba4
+-----------------------------------------------------------------------------------------------------------
+
+
+== Ensure that all tests are aware of the certificate and keys
+
+[source,shell]
+-----------------------------------------------------------------------------------------------------------
+cp ./ca.pem ../../../../../../../x-pack/plugin/security/src/test/resources/org/elasticsearch/xpack/security/authc/ldap/support/smb_ca.crt
+cp ./cert.pem ../../../../../../../x-pack/plugin/security/src/test/resources/org/elasticsearch/xpack/security/authc/ldap/support/smb_cert.crt
+cp ./cert.pem ../../../../../../../x-pack/plugin/core/src/test/resources/org/elasticsearch/xpack/security/transport/ssl/certs/simple/samba4.crt
+
+
+keytool -keystore ../../../../../../../x-pack/plugin/security/src/test/resources/org/elasticsearch/xpack/security/authc/ldap/support/ADtrust.jks -storepass changeit -delete -noprompt -alias smb_ca
+keytool -keystore ../../../../../../../x-pack/plugin/security/src/test/resources/org/elasticsearch/xpack/security/authc/ldap/support/ADtrust.jks -storepass changeit -importcert -file ca.pem -alias smb_ca -noprompt
+keytool -keystore ../../../../../../../x-pack/plugin/security/src/test/resources/org/elasticsearch/xpack/security/authc/ldap/support/ADtrust.jks -storepass changeit -delete -noprompt -alias smb_cert
+keytool -keystore ../../../../../../../x-pack/plugin/security/src/test/resources/org/elasticsearch/xpack/security/authc/ldap/support/ADtrust.jks -storepass changeit -importcert -file cert.pem -alias smb_cert -noprompt
+keytool -keystore ../../../../../../../x-pack/plugin/security/src/test/resources/org/elasticsearch/xpack/security/authc/ldap/support/ADtrust.jks -storepass changeit -delete -noprompt -alias mykey
+keytool -keystore ../../../../../../../x-pack/plugin/security/src/test/resources/org/elasticsearch/xpack/security/authc/ldap/support/ADtrust.jks -storepass changeit -importcert -file cert.pem -alias mykey -noprompt
```

Comment:
I've tested `UserAttributeGroupsResolverTests` and `ActiveDirectoryGroupsResolverTests` (the only users of `ADtrust.jks`) with only the `smb_ca` entry in the truststore (no `smb_cert` and no `mykey` entries) and they pass.

Might be worth adjusting the truststore while we're at it.

Reply:
Can do since I'll respin this eitherway, but we definitely need to clean our key/cert resources up soon. I think we have an open issue about this somewhere too